python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -225,7 +225,7 @@ def shutdown(self):
         from Ganga.Utility.logging import getLogger
         logger = getLogger()
         logger.debug("Shutting Down GangaRepositoryLocal: %s" % self.registry.name)
-        for k in self._fully_loaded.keys():
+        for k in [_k for _k in self._fully_loaded.keys() if _k not in self.incomplete_objects]:
```

Comment:
A belt and braces check here will just hide potentially large problems. If we think there could be a problem with the consistency of these two lists then we need to tackle that problem, not ignore it.